pkg/oauth/apiserver/registry/personaltokens/delegate/delegate.go

```diff
@@ -0,0 +1,186 @@
+package delegate
+
+import (
+	"context"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metainternal "k8s.io/apimachinery/pkg/apis/meta/internalversion"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	apirequest "k8s.io/apiserver/pkg/endpoints/request"
+	"k8s.io/apiserver/pkg/registry/rest"
+
+	"github.com/openshift/api/oauth"
+	oauthv1 "github.com/openshift/api/oauth/v1"
+
+	oauthapi "github.com/openshift/oauth-apiserver/pkg/oauth/apis/oauth"
+	accesstokenregistry "github.com/openshift/oauth-apiserver/pkg/oauth/apiserver/registry/oauthaccesstoken/etcd"
+	oauthprinters "github.com/openshift/oauth-apiserver/pkg/oauth/printers/internalversion"
+	"github.com/openshift/oauth-apiserver/pkg/printers"
+	"github.com/openshift/oauth-apiserver/pkg/printerstorage"
+)
+
+// rest implements a RESTStorage for access tokens against etcd
+type REST struct {
+	accessTokenStorage *accesstokenregistry.REST
+
+	tableConvertor rest.TableConvertor
+}
+
+var _ rest.Lister = &REST{}
+var _ rest.Getter = &REST{}
+var _ rest.Watcher = &REST{}
+var _ rest.GracefulDeleter = &REST{}
+var _ rest.Scoper = &REST{}
+
+// NewREST returns a RESTStorage object that will work against access tokens
+func NewREST(accessTokenStorage *accesstokenregistry.REST) (*REST, error) {
+	return &REST{
+		accessTokenStorage: accessTokenStorage,
+		tableConvertor:     printerstorage.TableConvertor{TableGenerator: printers.NewTableGenerator().With(oauthprinters.AddOAuthOpenShiftHandler)},
+	}, nil
+}
+
+func (r *REST) New() runtime.Object {
+	return &oauthv1.PersonalAccessToken{}
+}
+
+func (r *REST) NewList() runtime.Object {
+	return &oauthv1.PersonalAccessTokenList{}
+}
+
+func (r *REST) List(ctx context.Context, options *metainternal.ListOptions) (runtime.Object, error) {
+	ctxUserName, err := getUserFromContext(ctx)
```

Comment:
can it be empty?

Reply:
nvmd. It's your func below.

Reply:
`getUserFromContext` checks for an empty username and errors out in that case